lib/utils/date_utils.ts

```diff
@@ -112,10 +112,27 @@ function createDatePiece(year : number|null, month : number|null, day : number|n
     return date;
 }
 
-function createWeekDayDate(weekday : string, time : AbsoluteTime|null) {
-    const date = new Date;
+function weekdayToNumber(weekday : WeekDay) : number {
+    switch (weekday) {
+        case "monday": return 1;
+        case "tuesday": return 2;
+        case "wednesday": return 3;
+        case "thursday": return 4;
+        case "friday": return 5;
+        case "saturday": return 6;
+        case "sunday": return 7;
+    }
+    throw new Error(`Invalid weekday: ${weekday}`);
+}
 
-    // FIXME: implement this
+function createWeekDayDate(weekday : WeekDay, time : AbsoluteTime|null) {
+    const date = new Date;
+    const weekdayNumber = weekdayToNumber(weekday);
+    const diff = (weekdayNumber - date.getDate()) % 7;
+    // get the date of next specified weekday, today excluded
+    date.setDate(date.getDate() + (diff > 0 ? diff : diff + 7));
```

Comment:
I think this can be written more concisely as:
```
const diff = (7 + weekdayNumber - date.getDay()) % 7;
date.setDate(date.getDate() + diff);
```

(also, you have a typo there, getDate should be getDay in the diff calculation)

Reply:
lol, today is friday and 5th, so tests passed..

Reply:
the thing is sunday is 0, and I want to exclude today, that's why I didn't use that form

Reply:
oh i see, ok